scripts/run_example

```diff
@@ -80,8 +80,8 @@ if [[ "${server}" != "none" ]]; then
   # Run Backend for specific examples.
   if [[ "${EXAMPLE}" == 'aggregator' ]] || [[ "${EXAMPLE}" == 'trusted_information_retrieval' ]]; then
     # Building the Backend first, so it will not be built in the background.
-    cargo build --release --target=x86_64-unknown-linux-gnu --manifest-path="examples/${EXAMPLE}/backend/Cargo.toml"
-    cargo run --release --target=x86_64-unknown-linux-gnu --manifest-path="examples/${EXAMPLE}/backend/Cargo.toml" -- \
+    cargo build --release --manifest-path="examples/${EXAMPLE}/backend/Cargo.toml"
```

Comment:
FYI I think @rbehjati is trying to get to the bottom of whether specifying a `--target`, even if it is the same as the default, actually has any effect on caching.

Reply:
If it does have an effect, we could maybe have One True Setting in `scripts/common` so that macOS builds don't keep getting inadvertently broken.

Reply:
Specifying the target avoids invalidation of the cache. It is discussed extensively in #971. It might not be the best solution, but it is the best solution we currently have. Why do you think it should be removed @daviddrysdale?

Reply:
Specifying the target avoids invalidation of the cache. It is discussed extensively in #971. It might not be the best solution, but it is the best solution we currently have. Why do you think it should be removed @daviddrysdale?

Reply:
Explicitly specifying `x86_64-unknown-linux-gnu` prevents use of the scripts on other platforms, e.g. macOS and (presumably) ARM.

Reply:
@daviddrysdale What do you think about #1193? It gets the name of default target and uses that to set `--target`. 

The following change cannot be included in `scripts/common` because it uses `rustc` which is not installed in the android image, and causes the android-related build steps to fail. We could put it in a new `scripts/rust_common` file.
https://github.com/project-oak/oak/blob/fc1b92b3cfa260dd83249e64f46c31829da079eb/scripts/build_example#L8